xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingService.java

```diff
@@ -33,38 +63,71 @@ private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
   @Override
   protected String getChannelNameFromMessage(JsonNode message) {
     final String messageType = message.get("messageType").asText();
-    if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
+    if (messageType.startsWith(CHANNEL_ORDERBOOK)
+        | messageType.startsWith(CHANNEL_TICKER)
+        | messageType.startsWith(CHANNEL_TRADE)) {
       return messageType + ":" + message.get("marketId").asText();
     }
     return messageType;
   }
 
   @Override
   public String getSubscribeMessage(String channelName, Object... args) throws IOException {
-    if (CHANNEL_ORDERBOOK.equals(channelName)) {
-      subscribedOrderbooks.add(args[0].toString());
-      LOG.debug("Now subscribed to orderbooks {}", subscribedOrderbooks);
-      return objectMapper.writeValueAsString(buildSubscribeMessage());
+
```

Comment:
can you use an immutable set here and use the pattern found in `BitmexStreamingService.java` to look this up. Also single | is bitwise or instead of || so I don't think this would work the way you expect here.

Reply:
the example uses google classes as well. No changes made this round. The next iteration (expanding the messages consumed) will address this using a List but it is not immutable. Will look into the latter.